spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/elasticsearch/ElasticsearchProperties.java

```diff
@@ -62,6 +62,11 @@ public class ElasticsearchProperties {
 	 */
 	private String pathPrefix;
 
+	/**
+	 * Whether to enable keepalive between client and Elasticsearch.
+	 */
+	private boolean keepalive = true;
```

Comment:
I do think keep alive is important enough to be enabled by default to avoid sneaky troubles, however I also approve what you said: leave it to Elasticsearch itself, and change the default behavior when Elasticsearch's default changes.